src/neuroconv/basedatainterface.py

```diff
@@ -14,7 +14,7 @@ class BaseDataInterface(ABC):
     @classmethod
     def get_source_schema(cls):
         """Infer the JSON schema for the source_data from the method signature (annotation typing)."""
-        return get_schema_from_method_signature(cls.__init__, exclude=["source_data"])
+        return get_schema_from_method_signature(cls.__new__, exclude=["source_data"])
```

Comment:
the tests are failing so something else is missing here, any suggestions would be appreciated, not sure how to fix.